server/keyspace/keyspace_test.go

```diff
@@ -0,0 +1,275 @@
+	re.Error(err)
+}
+
+func makeMutations() []*keyspacepb.Mutation {
+	return []*keyspacepb.Mutation{
+		{
+			Op:    keyspacepb.Op_PUT,
+			Key:   []byte(testConfig1),
+			Value: []byte("new val"),
+		},
+		{
+			Op:    keyspacepb.Op_PUT,
+			Key:   []byte("new config"),
+			Value: []byte("new val"),
+		},
+		{
+			Op:  keyspacepb.Op_DEL,
+			Key: []byte(testConfig2),
+		},
+	}
+}
+
+func TestUpdateKeyspaceConfig(t *testing.T) {
+	re := require.New(t)
+	manager := newKeyspaceManager()
+	requests := makeCreateKeyspaceRequests(5)
+	for _, createRequest := range requests {
+		_, err := manager.CreateKeyspace(createRequest)
+		re.NoError(err)
+		mutations := makeMutations()
+		updated, err := manager.UpdateKeyspaceConfig(createRequest.Name, mutations)
+		re.NoError(err)
+		checkMutations(re, createRequest.InitialConfig, updated.Config, mutations)
+		// Changing config of a ARCHIVED keyspace is not allowed.
+		_, err = manager.UpdateKeyspaceState(createRequest.Name, keyspacepb.KeyspaceState_DISABLED, time.Now())
+		re.NoError(err)
+		_, err = manager.UpdateKeyspaceState(createRequest.Name, keyspacepb.KeyspaceState_ARCHIVED, time.Now())
+		re.NoError(err)
+		_, err = manager.UpdateKeyspaceConfig(createRequest.Name, mutations)
+		re.Error(err)
+	}
+}
+
+func TestUpdateKeyspaceState(t *testing.T) {
+	re := require.New(t)
+	manager := newKeyspaceManager()
+	requests := makeCreateKeyspaceRequests(5)
+	for _, createRequest := range requests {
+		_, err := manager.CreateKeyspace(createRequest)
+		re.NoError(err)
+		oldTime := time.Now()
+		// Archiving an ENABLED keyspace is not allowed.
+		_, err = manager.UpdateKeyspaceState(createRequest.Name, keyspacepb.KeyspaceState_ARCHIVED, oldTime)
+		re.Error(err)
+		// Disabling an ENABLED keyspace is allowed. Should update StateChangedAt.
+		updated, err := manager.UpdateKeyspaceState(createRequest.Name, keyspacepb.KeyspaceState_DISABLED, oldTime)
+		re.NoError(err)
+		re.Equal(updated.State, keyspacepb.KeyspaceState_DISABLED)
+		re.Equal(updated.StateChangedAt, oldTime.Unix())
+
+		newTime := time.Now()
+		// Disabling an DISABLED keyspace is allowed. Should NOT update StateChangedAt.
+		updated, err = manager.UpdateKeyspaceState(createRequest.Name, keyspacepb.KeyspaceState_DISABLED, newTime)
+		re.NoError(err)
+		re.Equal(updated.State, keyspacepb.KeyspaceState_DISABLED)
+		re.Equal(updated.StateChangedAt, oldTime.Unix())
+		// Archiving a DISABLED keyspace is allowed. Should update StateChangeAt.
+		updated, err = manager.UpdateKeyspaceState(createRequest.Name, keyspacepb.KeyspaceState_ARCHIVED, newTime)
+		re.NoError(err)
+		re.Equal(updated.State, keyspacepb.KeyspaceState_ARCHIVED)
+		re.Equal(updated.StateChangedAt, newTime.Unix())
+		// Changing state of an ARCHIVED keyspace is not allowed.
+		_, err = manager.UpdateKeyspaceState(createRequest.Name, keyspacepb.KeyspaceState_ENABLED, newTime)
+		re.Error(err)
+	}
+}
+
+func TestLoadRangeKeyspace(t *testing.T) {
```

Comment:
Suggest check that if the parameter range out of keyspaces actual range in storage, it will return keyspaces in actual range. And need consider the case that startID is greater than max(keyspaceID) in storage